internal/go_repository.bzl

```diff
@@ -147,7 +147,13 @@ def _go_repository_impl(ctx):
             print("fetch_repo: " + result.stderr)
 
     if generate:
-        # Build file generation is needed
+        # Build file generation is needed. Populate Gazelle directive at root build file
+        ctx.file(
+            ctx.attr.build_file_name.split(",")[0],
```

Comment:
It's possible that a build file exists with a different name, for example, if `build_file_generation = "on"` and a `BUILD` file is present. We should check this in the loop at L91, which will need to be unconditional.